docs/interpreter/spark.md

```diff
@@ -365,6 +370,52 @@ myScalaDataFrame = DataFrame(z.get("myScalaDataFrame"), sqlContext)
   </div>
 </div>
 
+### Object Interpolation
+Some interpreters can interpolate object values from `z` into the paragraph text by using the 
+`{variable-name}` syntax. The value of any object previously `put` into `z` can be 
+interpolated into a paragraph text by using such a pattern containing the object's name. 
+The following example shows one use of this facility:
+
+####In Scala cell:
+```
+z.put("minAge", 35)
+```
+
+####In later SQL cell:
+```
+%sql select * from members where age >= {minAge}
+```
+
+The interpolation of a `{var-name}` pattern is performed only when `z` contains an object with the specified name.
+But the pattern is left unchanged if the named object does not exist in `z`.
+Further, all `{var-name}` patterns within the paragraph text must must be translatable for any interpolation to occur -- 
+translation of only some of the patterns in a paragraph text is never done.
+
+In some situations, it is necessary to use { and } characters in a paragraph text without invoking the 
+object interpolation mechanism. For these cases an escaping mechanism is available -- 
+doubled braces {{ and }} should be used. The following example shows the use of {{ and }} for passing a 
+regular expression containing just { and } into the paragraph text.
+
+```
+%sql select * from members where name rlike '[aeiou]{{3}}'
+```
+
+To summarize, patterns of the form `{var-name}` within the paragraph text will be interpolated only if a predefined 
+object of the specified name exists. Additionally, all such patterns within the paragraph text should also 
+be translatable for any interpolation to occur. Patterns of the form `{{any-text}}` are translated into `{any-text}`. 
+These translations are performed only when all occurrences of `{`, `}`, `{{`, and `}}` in the paragraph text conform 
+to one of the two forms described above. Paragraph text containing `{` and/or `}` characters used in any other way 
+(than `{var-name}` and `{{any-text}}`) is used as-is without any changes. 
+No error is flagged in any case. This behavior is identical to the implementation of a similar feature in 
+Jupyter's shell invocation using the `!` magic command.
+
+This feature is disabled by default, and must be explicitly turned on for each interpreter by setting the value of 
+an interpreter-specific property to `true`. Consult the configuration section of each interpreter's documentation 
+to find out if object interpolation is implemented, and the name of the parameter that must be set to `true` to 
+enable the feature.
+
+At present only the SparkSQL and Shell interpreters support object interpolation. 
```

Comment:
so can the user uses the same variable in sql and then in shell?
let's add an example here?

Reply:
No, the parameter name is different for each interpreter as the implementation is per-interpreter, and the feature can be enabled/disabled independently for each interpreter. Have expanded the description and have included the parameter names for the two interpreters.

Reply:
In your question, if you meant _Z-variable_ name, then the answer is *yes*. There is an example added in both document files (for SparkSQL and Shell) illustrating this now.

Reply:
yes, for variable